backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -392,17 +406,18 @@ def grant_dataset_bucket_key_policy(self):
 
```

Comment:
I changed the way we were checking the `DATAALL_KMS_PIVOT_ROLE_PERMISSIONS_SID` because it was not updating the statement if it already existed